src/databricks/labs/ucx/hive_metastore/grants.py

```diff
@@ -318,18 +333,13 @@ def grants(
             any_file=any_file,
             anonymous_function=anonymous_function,
         )
-        try:  # pylint: disable=too-many-try-statements
-            grants = []
-            object_type_normalization = {
-                "SCHEMA": "DATABASE",
-                "CATALOG$": "CATALOG",
-                "ANY_FILE": "ANY FILE",
-                "ANONYMOUS_FUNCTION": "ANONYMOUS FUNCTION",
-            }
+        expected_grant_object_type = self._grants_reported_as.get(on_type, on_type)
+        grants = []
+        try:
             for row in self._fetch(f"SHOW GRANTS ON {on_type} {escape_sql_identifier(key)}"):
                 (principal, action_type, object_type, _) = row
-                object_type = object_type_normalization.get(object_type, object_type)
-                if on_type != object_type:
+                # This seems to be here to help unit tests that don't sufficiently narrow the target of mocked queries. -ajs
```

Comment:
Can we verify this comment and update it to be affirmative instead of questioning

Reply:
So that's interesting. As written (prior to this PR, and preserved as such) this piece of code filters out grants for object types that weren't requested while dealing with the fact that sometimes grants are reported slightly differently to how they were requested.

The current code is from here, squashed from #246: https://github.com/databrickslabs/ucx/blob/cbc26542f2c7edfcfe0e9d193dfe01803acb1333/src/databricks/labs/ucx/hive_metastore/grants.py#L231

And there was some discussion about it [here](https://github.com/databrickslabs/ucx/pull/246#discussion_r1331907633) where @william-conti wrote:
> It's a corner case where the object type isn't the same as the one requested. For example you request the grant for a view and you get the grants for a table, it's not a normal behaviour and should be highlighted

However this only addresses expected cases where the requested/reported types don't match: what we don't know is when in practice this actually occurs (and why the guard is in place).

Since the original code was written, it turns out that the main place this check is used is with unit tests written in the meantime that don't mock the statements correctly. Essentially these mock `SHOW GRANTS ON .*` and therefore the mocks report table grants when we query the catalog grants and stuff like that. Here's an example of this:
https://github.com/databrickslabs/ucx/blob/a8e823f1bcff971f7bfa266bb4233216fb8dc4ac/tests/unit/hive_metastore/test_grants.py#L205

As things stand instead of this check I think I'd prefer to fix the unit tests (they're unintentionally mocking "broken" behaviour) and remove the check entirely.

Reply:
Agreed, but out-of-scope for this PR though. Could you create a tech debt issue for this?

Reply:
> Could you create a tech debt issue for this?

Done; #1932.